src/structure.js

```diff
@@ -277,15 +280,14 @@ Structure.prototype.reference = function (path) {
      * @returns {Void}
      */
     destroy: function () {
-      removeAllListenersBut(self, pathId, referenceListeners);
-      referenceListeners = void 0;
-      cursor = void 0;
+      removeAllListenersBut(listenerNs);
+      cursor = undefined;
 
       this._dead = true;
-      this.observe = void 0;
-      this.unobserveAll = void 0;
-      this.cursor = void 0;
-      this.destroy = void 0;
+      this.observe = undefined;
```

Comment:
You can git reset soft to go back and discard hunks by doing 'git checkout -p file'. You can force push changes to a PR, so no need to worry about the history.